components/applications-service/pkg/storage/postgres/hab_event.go

```diff
@@ -134,41 +134,167 @@ func (db *Postgres) IngestHealthCheckEventWithoutMetrics(event *habitat.HealthCh
 
 	// @afiune all our backend was designed for the health check to be all
 	// uppercases but habitat is actually sending case sensitive strings
-	newHealth := strings.ToUpper(event.GetResult().String())
+	eventHealth := strings.ToUpper(event.GetResult().String())
 	// TODO @afiune verify health
 
-	// If the service already exists, we just do a simple update
+	// If the service already exists, we just do an update
 	if exist {
-		return db.updateService(
+		return db.updateTables(
 			svc,
 			eventMetadata,
 			svcMetadata,
 			pkgIdent,
-			newHealth,
+			eventHealth,
 		)
 	}
 
 	// But if the service doesn't exist, we will handle it as a new service insertion
-	err1 := db.insertNewService(
+	err = db.insertNewService(
 		eventMetadata,
 		svcMetadata,
 		pkgIdent,
-		strings.ToUpper(event.GetResult().String()),
+		eventHealth,
 	)
 
-	if err1 != nil {
+	if err != nil {
 		// We will retry once if a unique constraint is hit,
 		// in case a new deployment or service group is in the same batch of events.
-		if err1.Error() == uniqueDeploymentError || err1.Error() == uniqueServiceGroupError {
+		if err.Error() == uniqueDeploymentError || err.Error() == uniqueServiceGroupError {
 			return db.insertNewService(
 				eventMetadata,
 				svcMetadata,
 				pkgIdent,
-				strings.ToUpper(event.GetResult().String()),
+				eventHealth,
 			)
 		}
 	}
-	return err1
+	return err
+}
+
+func (db *Postgres) updateTables(
+	svc *service,
+	eventMetadata *habitat.EventMetadata,
+	svcMetadata *habitat.ServiceMetadata,
+	pkgIdent *packageIdent,
+	health string) error {
+
+	//updates := map[string]interface{}{}
```

Comment:
leftover cruft?

Reply:
Oh yeah, thanks for the reminder!! I thought about making the functions a bit smarter and simpler by passing an array of tables to be updated, I was in the middle of it and forgot to finish it! I'll finish it and push my idea.

Reply:
here: https://github.com/chef/automate/pull/779/commits/c82b8ba0efba724c140721605397e9ef4abe7e1e check it out!

![tenor-250846682](https://user-images.githubusercontent.com/5712253/60873897-68671d80-a237-11e9-81df-5bd2900aff0a.gif)